src/pybind/rados/rados.pyx

```diff
@@ -3937,6 +3937,44 @@ returned %d, but should return zero on success." % (self.name, ret))
         finally:
             free(apps)
 
+    def application_metadata_get(self, app_name, key):
+        """
+        Gets application metadata on an OSD pool for the given key
+
+        :param app_name: application name
+        :type app_name: str
+        :param key: metadata key
+        :type key: str
+        :returns: str - metadata value
+
+        :raises: :class:`Error`
+        """
+
+        app_name =  cstr(app_name, 'app_name')
+        key = cstr(key, 'key')
+        cdef:
+            char *_app_name = app_name
+            char *_key = key
+            size_t size = 4096
```

Comment:
the max_application_data_length is hard-coded at 128, so 128 + 1 would suffice, see https://github.com/ceph/ceph/blob/master/src/mon/OSDMonitor.cc#L231

Reply:
done, thx.